code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -100,11 +118,16 @@
 
 /obj/machinery/atmospherics/components/binary/thermomachine/update_overlays()
 	. = ..()
-	. += getpipeimage(icon, "pipe", dir, COLOR_LIME, piping_layer)
-	. += getpipeimage(icon, "pipe", turn(dir, 180), COLOR_MOSTLY_PURE_RED, piping_layer)
+	if(!initial(icon))
+		return
+	var/mutable_appearance/thermo_overlay = new(initial(icon))
```

Comment:
Could this be cached?

Reply:
I put the blinking overlay to use the GAGS JSON icon state instead with d3651b879263d290edebe1064f7523a9a2ad06dc

Reply:
My question still applies, we're making an I think? identical mutable appearance for each thermomachine, could we cache it?

Reply:
I missed your response.  The mutable appearance is for the pipe overlays which is something my PR doesn't touch.  I am not familiar with caching the overlays in DM.

If you are worried about performance, please consider that there are only ~5 thermomachines on a map so the speed increase would be _extremely meager_.  I have cached results for other code modules outside of DM & SS13, and when I benchmarked the results it's only worth it if there is a severe bottleneck in performance.  If there was something like thousands of instances, or a piece of code that was executed frequently then I could justify trying to boost the performance.

From my understanding of the mutable_apperance of the pipes below the thermomachine, it would have to cover all the pipe layers, directions, and colors.  I used the `!initial(icon)` code from the GAGS involving PDA's.  I'm still not even sure how it all works precisely.

Reply:
if I'm honest it's just a gut reaction from seeing an overlay being created often. You're right, caching isn't a one size fits all tool, and this is more of a pattern thing. Maybe worth doing, but if you're uncomfortable with it we can skip for now

Reply:
I'm going to opt to skip it.  Other than that, I believe everything else in this PR is ready to go.